### PR TITLE
DCMAW-9802: Refactor SSM Service functionality into client credentials service

### PR DESCRIPTION
**Pull request by J-son1**
​DCMAW-9802
### What changed
- Remove `SSMService` and refactored functions into `ClientCredentialsService`
- Created `SSMClient` as this was previous done in the SSMService and didn't want this to live in the `ClientCredentialsService` either

### Why did it change
No need to have both the `SSMService` and  `ClientCredentialsService` as `SSMService` was using SSM to get client credentials and then validating them. We decided it should be the responsibility of the `ClientCredentialsService` to do this.

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [x] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [ ] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: backend-api/src/functions/asyncCredential/asyncCredentialHandler.test.ts [hunk 1]
```diff
@@ -627,145 +597,112 @@ describe("Async Credential", () => {
...
-            sub: "mockSub",
-            client_id: "mockClientId",
-            govuk_signin_journey_id: "mockGovukSigninJourneyId",
-          }),
-        });
-
-        dependencies.clientCredentialsService = () =>
-          new MockPassingClientCredentialsServiceInvalidIssuer();
+    describe("Validate client credentials", () => {
+      describe("Given redirect_uri is not valid", () => {
```
Comment (jamesamooney) [inline backend-api/src/functions/asyncCredential/asyncCredentialHandler.test.ts:674]: This should not longer be case specific, will update

Comment (jamesamooney) [reply backend-api/src/functions/asyncCredential/asyncCredentialHandler.test.ts]: Updated

File: backend-api/src/functions/services/clientCredentialsService/clientCredentialsService.ts [hunk 2]
```diff
@@ -1,8 +1,71 @@
...
+    cache = {
+      expiry: Date.now() + this.cacheTTL,
+      data: parsedCredentials,
+    };
+
+    return successResult(parsedCredentials);
+  };
 
-export class ClientCredentialsService implements IClientCredentialsService {
   validateTokenRequest = (
```
Comment (jamesamooney) [inline backend-api/src/functions/services/clientCredentialsService/clientCredentialsService.ts:69]: Should we rename this to `validateAsyncTokenRequest` to align with `validateAsyncCredentialRequest`

Comment (jamesamooney) [reply backend-api/src/functions/services/clientCredentialsService/clientCredentialsService.ts]: Done

**Review by jamesamooney — DISMISSED** [review]

File: backend-api/src/functions/asyncCredential/asyncCredentialHandler.ts [hunk 3]
```diff
@@ -360,8 +336,10 @@ export interface Dependencies {
   logger: () => Logger<MessageName>;
   eventService: (sqsQueue: string) => IEventService;
   tokenService: () => IDecodeToken & IVerifyTokenSignature;
-  clientCredentialsService: () => ClientCredentialsService;
-  ssmService: () => IGetClientCredentials;
+  clientCredentialsService: () => IGetClientCredentials &
+    IValidateTokenRequest &
```
Comment (jamesamooney) [inline backend-api/src/functions/asyncCredential/asyncCredentialHandler.ts:340]: Update name now func name has changed

File: backend-api/src/functions/services/clientCredentialsService/clientCredentialsService.test.ts [hunk 4]
```diff
@@ -200,7 +395,7 @@ describe("Client Credentials Service", () => {
     });
 
     describe("Given the supplied credential clientId is present in the stored credentials array", () => {
```
Comment (J-son1) [inline backend-api/src/functions/services/clientCredentialsService/clientCredentialsService.test.ts:397]:
```suggestion
    describe("Given the supplied clientId is present in the stored credentials array", () => {
```

File: backend-api/src/functions/asyncCredential/asyncCredentialHandler.ts (hunk 3 above)
Comment (jamesamooney) [reply backend-api/src/functions/asyncCredential/asyncCredentialHandler.ts]: Updated

**Review by jamesamooney — APPROVED** [review]

**Pull request merged**